d2l-visible-on-ancestor-behavior.html

```diff
@@ -0,0 +1,119 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="d2l-dom.html">
+<link rel="import" href="../d2l-fastdom-import/fastdom.html">
+
+<script>
+	window.D2L = window.D2L || {};
+	window.D2L.PolymerBehaviors = window.D2L.PolymerBehaviors || {};
+
+	/** @polymerBehavior */
+	D2L.PolymerBehaviors.VisibleOnAncestorBehavior = {
+
+		properties: {
+
+			/**
+			 * Whether the element is only visible when hovered or focus is within ancestor with 'd2l-visible-on-ancestor-target'.
+			 */
+			visibleOnAncestor: {
+				type: Boolean,
+				reflectToAttribute: true
+			}
+
+		},
+
+		attached: function() {
+
+			if (!this.visibleOnAncestor) return;
+			this.__voaTarget = D2L.Dom.findComposedAncestor(this, function(node) {
+				if (!node || node.nodeType !== 1) return false;
```

Comment:
The `classList` reference on the next line will bomb on the document element, so we need to check for element here first.